orchagent/bfdorch.cpp

```diff
@@ -96,18 +106,66 @@ void BfdOrch::doTask(Consumer &consumer)
 
         if (op == SET_COMMAND)
         {
-            if (!create_bfd_session(key, data))
+            bool tsa_shutdown_enabled = false;
+            for (auto i : data)
             {
-                it++;
-                continue;
+                auto value = fvValue(i);
+                //shutdown_bfd_during_tsa parameter is used by the BFD session creator to ensure that the the
+                //specified session gets removed when the device goes into TSA state.
+                //if this parameter is not specified or set to false for a session, the
+                // corrosponding BFD session would be maintained even in TSA state.
+                if (fvField(i) == "shutdown_bfd_during_tsa" && value == "true" )
```

Comment:
@arlakshm , fyi (This param can control the individual BFD session state during TSA)